pkg/mcp/server_test.go

```
@@ -0,0 +1,86 @@
+	go func() {
+		_ = server.Start()
+	}()
+
+	// Give the server a moment to start.
+	time.Sleep(200 * time.Millisecond)
+
+	baseURL := fmt.Sprintf("http://localhost%s", cfg.Url)
+
+	// /healthz must be reachable without a token.
+	t.Run("unauthenticated /healthz", func(t *testing.T) {
+		resp, err := http.Get(baseURL + "/healthz")
```

golangci-lint (2.12.2)

[error] net/http.Get must not be called. use net/http.NewRequestWithContext and (*net/http.Client).Do(*http.Request)

(noctx)